src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs

```diff
@@ -241,7 +241,38 @@ int IList.Add(object? item)
         // capacity or the new size, whichever is larger.
         //
         public void AddRange(IEnumerable<T> collection)
-            => InsertRange(_size, collection);
+        {
+            if (collection == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.collection);
+            }
+
+            if (collection is ICollection<T> c)
+            {
+                int count = c.Count;
+                if (count > 0)
+                {
+                    if (_items.Length - _size < count)
+                    {
+                        Grow(_size + count);
+                    }
+
+                    c.CopyTo(_items, _size);
+                    _size += count;
+                    _version++;
+                }
+            }
+            else
+            {
+                using (IEnumerator<T> en = collection.GetEnumerator())
+                {
+                    while (en.MoveNext())
+                    {
+                        Add(en.Current);
+                    }
+                }
```

Comment:
```suggestion
                foreach (T item in collection)
                {
                    Add(item);
                }
```
Aren't these two equivalent?

Reply:
The [generated IL](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAGAAkIEYUBuAWACgTyKA6AJQFcA7DbAWxgsAwhD4AHPDCgBlKQDdsYGAGc6jMtIAWAQyhiAMtuDtuvAWoYBtAFLYMAcRhcpigBQYAnmJgQAZq+weAEoggF0GQgBmcgAmUgAVFQwhXG1lZQAeeIA+BgBvBlIi8mjKJHIUUgBBABMatm0uAHMYV0pIrOzSSFxcGDBeCC4gwuKC+mLJ0l9oGG0wTVJXeNI7GD5Vrm6IXv7B4eq6gIx1oLVJgF9RouuS5nLCStr6xpaYtooOnO3dgewhkYTMa3SZMVwASQAolwOAIoNoMNBOqQnKQALw/Pp/IYsRwYaGwqQI6CuEIg4FAqZTADumkkSycLAAshA5DAAHIwBAYUmAqlU8b8oWHGquRlCDhQWDBc7CopXSlTBWXW63SxMmAYTQQGrg8S4VwarU6vViXAAeTE+2ULHZEHBXFwgUCTTCtyi9wqIuWqxOfCCpDyCouQA===) looks the same in this case since the current isn't being captured.

Reply:
> The [generated IL](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAGAAkIEYUBuAWACgTyKA6AJQFcA7DbAWxgsAwhD4AHPDCgBlKQDdsYGAGc6jMtIAWAQyhiAMtuDtuvAWoYBtAFLYMAcRhcpigBQYAnmJgQAZq+weAEoggF0GQgBmcgAmUgAVFQwhXG1lZQAeeIA+BgBvBlIi8mjKJHIUUgBBABMatm0uAHMYV0pIrOzSSFxcGDBeCC4gwuKC+mLJ0l9oGG0wTVJXeNI7GD5Vrm6IXv7B4eq6gIx1oLVJgF9RouuS5nLCStr6xpaYtooOnO3dgewhkYTMa3SZMVwASQAolwOAIoNoMNBOqQnKQALw/Pp/IYsRwYaGwqQI6CuEIg4FAqZTADumkkSycLAAshA5DAAHIwBAYUmAqlU8b8oWHGquRlCDhQWDBc7CopXSlTBWXW63SxMmAYTQQGrg8S4VwarU6vViXAAeTE+2ULHZEHBXFwgUCTTCtyi9wqIuWqxOfCCpDyCouQA===) looks the same in this case since the current isn't being captured.

Try making your methods [instance instead of static](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAGAAkIEYUBuAWACgTyKA6AJQFcA7DbAWxgsAwhD4AHPDCgBlKQDdsYGAGc6jMtIAWAQyhiAMtuDtuvAWoYBtAFLYMAcRhcpigBQYAnmJgQAZq+weAEoggF0GQgBmcgAmUgAVFQwhXG1lZQAeeIA+BgBvBlIi8mjCFFIAQQATKrZtLgBzGFdKSKzs0khcXBgwXgguIMLigvpi8dJfaBhtME1SV3jSOxg+Za5OiG7e/sHKmoCMVaC1cYBfYaLLkvJy6tr6ppiWijacze2+7AGhsZHr8ZMVwASQAolwOAIoNoMNB2qQnKQALwfHpfAYsRwYcGQqQw6CuEIA/5/CYTADumkkCycLAAshA5DAAHIwBAYQm/Mlk0bcvn7KquWlCDhQWDBU78ooXUkTGXna7XSx0mAYTQQKrA8S4VwqtUarViXAAeTEu2ULGZEGBXFwgUCDTC1yitwFi2WRz4QVIeRlZyAA==).